smarts/core/agent_manager.py

```diff
@@ -102,7 +102,7 @@ def agent_interfaces(self) -> Dict[str, AgentInterface]:
         """A list of all agent to agent interface mappings."""
         return self._agent_interfaces
 
-    def agent_interface_for_agent_id(self, agent_id) -> AgentInterface:
+    def agent_interface_for_agent_id(self, agent_id) -> Optional[AgentInterface]:
         """Get the agent interface of a specific agent."""
         return self._agent_interfaces.get(agent_id, None)
```

Comment:
It looks like `agent_interface_for_agent_id` function is not expected to return `None`, hence consider the following.

```suggestion
    def agent_interface_for_agent_id(self, agent_id) -> AgentInterface:
        """Get the agent interface of a specific agent."""
        return self._agent_interfaces[agent_id]
```